pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/embeddings/HuggingFaceRestEmbeddingService.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.functions.transforms.embeddings;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
```

Comment:
What is the minimum Java version we shall support ?
If it's 11 then we could use the new HttpClient

Reply:
JDK11 is the base.

if we use HttpClient than probably it makes sense to cache it in a field, it will allow better performances (and it also supports HTTP2)

+1